ldes-server-infra-mongo/mongo-fetch-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetch/AllocationMongoRepository.java

```diff
@@ -34,6 +49,17 @@ public List<MemberAllocation> getMemberAllocationsByFragmentId(String fragmentId
 				.toList();
 	}
 
+	@Override
+	public List<String> getMemberAllocationIdsByFragmentIds(Set<String> fragmentIds) {
+		Query q = new Query();
+		q.addCriteria(Criteria.where("fragmentId").in(fragmentIds))
+				.fields().include("memberId");
+		return mongoTemplate.find(q, MemberAllocationEntity.class)
```

Comment:
Since you already use ".include("memberId")", maybe you can go all the way and exclude _id. Then you would be able to write this to a String.class and skip the stream->map->toList below.